packages/localstorage-adapter/modules/adapter/adapter.ts

```diff
@@ -114,22 +115,25 @@ export const updateFlags = (flags: TFlags) => {
   adapterState.emitter.emit('flagsStateChange', nextFlags);
 };
 
+const didFlagsChange = (nextFlags: TFlags) => {
+  const previousFlags = adapterState.flags;
+
+  if (previousFlags === undefined) return true;
+
+  return isEqual(nextFlags, previousFlags);
```

Comment:
should be a `!isEqual`, because the `true` from this function is true if the two flags are _not_ equal.